plugins/CoreAdminHome/API.php

```diff
@@ -107,4 +114,37 @@ public function runCronArchiving()
         $archiver = new CronArchive();
         $archiver->main();
     }
+
+    /**
+     * Ensure the specified dates are valid.
+     * Store invalid date so we can log them
+     * @param array $dates
+     * @return Date[]
+     */
+    private function getDatesToInvalidateFromString($dates)
```

Comment:
Right now just simplifying the invalidator to make it easier for me to change. Maybe I will create a DatesList class as we talked about re: https://github.com/piwik/piwik/issues/8852 if I end up w/ some time.